### PR TITLE
Handle wrong page params.

### PR DESCRIPTION
**Pull request by peregrineshahin**
fixes: https://github.com/glinscott/fishtest/issues/1636

### CODE REVIEW

File: server/fishtest/views.py [hunk 1]
```diff
@@ -1434,7 +1435,8 @@ def tests(request):
             ("Expires", "0"),
         )
     )
-    if int(request.params.get("page", 1)) > 1:
+    page_param = request.params.get("page", "")
+    if not page_param.isdigit() or int(page_param) > 1:
```
Comment (ppigazzini) [inline server/fishtest/views.py:1439]:
```diff
diff --git a/server/fishtest/views.py b/server/fishtest/views.py
index cb3fd7e..d1443fd 100644
--- a/server/fishtest/views.py
+++ b/server/fishtest/views.py
@@ -1317,7 +1317,8 @@ def get_paginated_finished_runs(request):
     yellow_only = request.params.get("yellow_only", False)
     ltc_only = request.params.get("ltc_only", False)

-    page_idx = max(0, int(request.params.get("page", 1)) - 1)
+    page_param = request.params.get("page", "")
+    page_idx = max(0, int(page_param) - 1 if page_param.isdigit() else 0)
     page_size = 25

     finished_runs, num_finished_runs = request.rundb.get_finished_runs(
@@ -1374,7 +1375,8 @@ def tests_user(request):
     )
     username = request.matchdict.get("username", "")
     response = {**get_paginated_finished_runs(request), "username": username}
-    if int(request.params.get("page", 1)) == 1:
+    page_param = request.params.get("page", "")
+    if not page_param.isdigit() or int(page_param) <= 1:
         response["runs"] = request.rundb.aggregate_unfinished_runs(
             username=username,
         )[0]
@@ -1434,7 +1436,8 @@ def tests(request):
             ("Expires", "0"),
         )
     )
-    if int(request.params.get("page", 1)) > 1:
+    page_param = request.params.get("page", "")
+    if page_param.isdigit() and int(page_param) > 1:
         # page 2 and beyond only show finished test results
         return get_paginated_finished_runs(request)
```

Comment (peregrineshahin) [reply server/fishtest/views.py:1439]: Are you sure about the last one
```
+    page_param = request.params.get("page", "")
+    if page_param.isdigit() and int(page_param) > 1:
```
this check for isdigit appears to be redundant, so no need for it right?

Comment (ppigazzini) [reply server/fishtest/views.py:1440]: The proposed code deals with "https://dfts-0.pigazzini.it/tests?page=qfdfasd" as well. I thought that that was your intention in using `isdigit()`

Comment (ppigazzini) [reply server/fishtest/views.py:1440]: BTW nn pagination needs the same changes.
"https://dfts-0.pigazzini.it/nns?page=sdcsdjbddve" crash

Comment (peregrineshahin) [reply server/fishtest/views.py]: Makes sense pushed.

**Comment by peregrineshahin** [comment]
Pushed the requested changes but with
`+    page_idx = max(0, int(page_param) - 1) if page_param.isdigit() else 0`
instead of 
`+    page_idx = max(0, int(page_param) - 1 if page_param.isdigit() else 0`)

**Comment by ppigazzini** [comment]
Events log is paginated as well.
```diff
@@ -377,7 +378,9 @@ def actions(request):
     if max_actions:
         max_actions = int(max_actions)

-    page_idx = max(0, int(request.params.get("page", 1)) - 1)
+    page_param = request.params.get("page", "")
+    page_idx = max(0, int(page_param) - 1) if page_param.isdigit() else 0
+
     page_size = 25
```

**Comment by ppigazzini** [comment]
You have missed the required change in `get_paginated_finished_runs()` in my first diff :)

```diff
diff --git a/server/fishtest/views.py b/server/fishtest/views.py
index cb3fd7e..e1386f2 100644
--- a/server/fishtest/views.py
+++ b/server/fishtest/views.py
@@ -301,7 +301,8 @@ def nns(request):
     network_name = request.params.get("network_name", "")
     master_only = request.params.get("master_only", False)

-    page_idx = max(0, int(request.params.get("page", 1)) - 1)
+    page_param = request.params.get("page", "")
+    page_idx = max(0, int(page_param) - 1) if page_param.isdigit() else 0
     page_size = 25

     nns_list, num_nns = request.rundb.get_nns(
@@ -377,7 +378,9 @@ def actions(request):
     if max_actions:
         max_actions = int(max_actions)

-    page_idx = max(0, int(request.params.get("page", 1)) - 1)
+    page_param = request.params.get("page", "")
+    page_idx = max(0, int(page_param) - 1) if page_param.isdigit() else 0
+
     page_size = 25

     actions, num_actions = request.actiondb.get_actions(
@@ -1317,7 +1320,8 @@ def get_paginated_finished_runs(request):
     yellow_only = request.params.get("yellow_only", False)
     ltc_only = request.params.get("ltc_only", False)

-    page_idx = max(0, int(request.params.get("page", 1)) - 1)
+    page_param = request.params.get("page", "")
+    page_idx = max(0, int(page_param) - 1) if page_param.isdigit() else 0
     page_size = 25

     finished_runs, num_finished_runs = request.rundb.get_finished_runs(
@@ -1374,7 +1378,8 @@ def tests_user(request):
     )
     username = request.matchdict.get("username", "")
     response = {**get_paginated_finished_runs(request), "username": username}
-    if int(request.params.get("page", 1)) == 1:
+    page_param = request.params.get("page", "")
+    if not page_param.isdigit() or int(page_param) <= 1:
         response["runs"] = request.rundb.aggregate_unfinished_runs(
             username=username,
         )[0]
@@ -1434,7 +1439,8 @@ def tests(request):
             ("Expires", "0"),
         )
     )
-    if int(request.params.get("page", 1)) > 1:
+    page_param = request.params.get("page", "")
+    if page_param.isdigit() and int(page_param) > 1:
         # page 2 and beyond only show finished test results
         return get_paginated_finished_runs(request)
```

**Comment by peregrineshahin** [comment]
Done.

**Review by ppigazzini — APPROVED** [review]
LGTM on DEV

**Comment by ppigazzini** [comment]
PROD updated, thank you @peregrineshahin :)

**Pull request merged**